### PR TITLE
[RFC] Add lr into metric logging and also rename the loss name

### PR DESCRIPTION
**Pull request by fduwjj**
During some experiments we realized that we need to log lr into metrics so that we can ensure things are correct.

somehow the name of loss does not match with the run we want to compare. So I am wondering if we can build a metric update util so that we can set and change it for different workload?

For example changing from loss_metrics/global_avg_loss to train/avg_loss_AVG

### CODE REVIEW

**Review by tianyu-l — COMMENTED** [review]
fyi it seems similar things (with bigger change) are being done in https://github.com/pytorch/torchtitan/pull/938

Maybe we can suggest changes over there?

**Comment by fduwjj** [comment]
@tianyu-l this is different. Mostly this is for metric improvement. And we also need cosine lr as well (I just haven't got time to do that but it is ok) To me these are separate topic and the author is using lr_min.

File: torchtitan/train.py [hunk 1]
```diff
@@ -385,10 +386,20 @@ def main(job_config: JobConfig):
                 time_data_loading_pct = 100 * sum(data_loading_times) / time_delta
 
                 device_mem_stats = device_memory_monitor.get_peak_stats()
+                curr_lr = float(
+                    max(
```
Comment (wconstab) [inline torchtitan/train.py:390]: its a little unclear to me that 'max' is the right thing to do here.  if lr is different between param groups, wouldn't you want to log the different ones? or ...

File: torchtitan/tools/metrics.py [hunk 2]
```diff
@@ -96,6 +96,11 @@ def build_device_memory_monitor():
     return device_memory_monitor
 
 
+def metric_processor(metrics: Dict[str, Any]):
```
Comment (wconstab) [inline torchtitan/tools/metrics.py:99]: hmm.  maybe we can do this offline in a util if we want to rename metrics?  (i assume that's what this is for?)

Comment (fduwjj) [reply torchtitan/tools/metrics.py:99]: Well this is to update it offline but we need a hook in train.py.

File: torchtitan/train.py (hunk 1 above)
Comment (fduwjj) [reply torchtitan/train.py:390]: This is one definition of lr used in internal workload. It is for feature parity.

Comment (tianyu-l) [reply torchtitan/train.py:390]: IIUC currently in torchtitan, all `optimizers` and `param_groups` would have the same LR across all parameters. Could you elaborate why this is needed?

File: torchtitan/tools/metrics.py (hunk 2 above)
Comment (wconstab) [reply torchtitan/tools/metrics.py:99]: what i meant by offline is, run a script that processes the .tb file.  I don't know if changing the name of the written metrics is a common enough use case to justify adding a hook into torchtitan.  Perhaps its OK to just use some inconvenient methods to generate the comparison plots between torchtitan and an incompatible trainer?  idk... i see why you want this though.

**Comment by fduwjj** [comment]
Will adopt what is done here: https://github.com/pytorch/torchtitan/pull/945
